src/codecs/rotate/processor.ts

```diff
@@ -1,73 +1,26 @@
-import { RotateOptions } from './processor-meta';
+import wasmUrl from '../../../codecs/rotate/rotate.wasm';
+import { RotateOptions, RotateModuleInstance } from './processor-meta';
 
-export function rotate(data: ImageData, opts: RotateOptions): ImageData {
-  const { rotate } = opts;
-  const flipDimensions = rotate % 180 !== 0;
-  const { width: inputWidth, height: inputHeight } = data;
-  const outputWidth = flipDimensions ? inputHeight : inputWidth;
-  const outputHeight = flipDimensions ? inputWidth : inputHeight;
-  const out = new ImageData(outputWidth, outputHeight);
-  let i = 0;
+export async function rotate(
+  data: ImageData,
+  opts: RotateOptions,
+): Promise<ImageData> {
+  const flipDimensions = opts.rotate % 180 !== 0;
+  // Number of wasm memory pages (á 64KiB) needed to store the image twice.
+  const bytesPerImage = data.width * data.height * 4;
+  const numPagesNeeded = Math.ceil(bytesPerImage * 2 / (64 * 1024));
+  const { instance } = (await (WebAssembly as any).instantiateStreaming(
+    fetch(wasmUrl),
+  )) as { instance: RotateModuleInstance };
```

Comment:
Our other wasm modules can be used many times per instantiation. Do we want to do the same here?